### ISSUE TITLE
[Support]: Deleting Events take a long time

### ISSUE THREAD

**Issue by MasterOfPuppets111** [blakeblackshear/frigate#7565]
### Describe the problem you are having

For a few days, deleting events has been taking around 40-50 seconds. During this time, Frigate will not respond to user interface inputs. Before I could delete the events one after the other without delay, now I have to wait about half a minute for each event before I can delete the next one.
The log looks clean to me, no errors.
Frigate doesn't crash either.

### Version

0.12.1-367D724

### Frigate config file

```yaml
mqtt:
  enabled: false
  host: mqtt.server.com

detectors:
 coral:
  type: edgetpu
  device: usb

ui:
  use_experimental: false
  live_mode: mse

go2rtc:
  streams:
    holzlager:
      - rtsp://admin:XXX@192.168.1.10:554/Streaming/Channels/101
    holzlager_sub:
      - rtsp://admin:XXX@192.168.1.10:554/Streaming/Channels/102
    werkstatt:
      - rtsp://admin:XXX@192.168.1.11:554/Streaming/Channels/101
    werkstatt_sub:
      - rtsp://admin:XXX@192.168.1.11:554/Streaming/Channels/102
    haustuer_strasse:
      - rtsp://admin:XXX@192.168.1.12:554/Streaming/Channels/101
    haustuer_strasse_sub:
      - rtsp://admin:XXX@192.168.1.12:554/Streaming/Channels/102

cameras:
    holzlager:
      ffmpeg:
        hwaccel_args: preset-vaapi
        output_args:
          record: preset-record-generic-audio-copy
        inputs:
          #Hoch
          - path: rtsp://172.17.0.2:8554/holzlager
            roles:
              - record
          #Mittel
          - path: rtsp://172.17.0.2:8554/holzlager_sub
            roles:
              - detect
      detect:
        width: 1280
        height: 720
        fps: 5
      objects:
        track:
          - person
          - motorcycle
          - dog
          - cat
          - bird
      record:
        enabled: True
        events:
          pre_capture: 30
          post_capture: 30
          retain:
            default: 365
      snapshots:
        enabled: True
        retain:
            default: 365
      live:
        stream_name: holzlager
        quality: 1  

    werkstatt:
      ffmpeg:
        hwaccel_args: preset-vaapi
        output_args:
          record: preset-record-generic-audio-copy
        inputs:
          #Hoch
          - path: rtsp://172.17.0.2:8554/werkstatt
            roles:
              - record
          #Mittel
          - path: rtsp://172.17.0.2:8554/werkstatt_sub
            roles:
              - detect
      detect:
        width: 1280
        height: 720
        fps: 5
      objects:
        track:
          - person
          - motorcycle
          - dog
          - cat
          - bird
      record:
        enabled: True
        events:
          pre_capture: 30
          post_capture: 30
          retain:
            default: 365
      snapshots:
        enabled: True
        retain:
            default: 365
      live:
        stream_name: werkstatt
        quality: 1  

    haustuer_strasse:
      ffmpeg:
        hwaccel_args: preset-vaapi
        output_args:
          record: preset-record-generic-audio-copy
        inputs:
          #Hoch
          - path: rtsp://172.17.0.2:8554/haustuer_strasse
            roles:
              - record
          #Mittel
          - path: rtsp://172.17.0.2:8554/haustuer_strasse_sub
            roles:
              - detect
      detect:
        width: 720
        height: 1280
        fps: 5
      objects:
        track:
          - person
          - motorcycle
          - dog
          - cat
          - bird
      record:
        enabled: True
        events:
          pre_capture: 30
          post_capture: 30
          retain:
            default: 365
      snapshots:
        enabled: True
        retain:
            default: 365
      live:
        stream_name: haustuer_strasse
        quality: 1  
      motion:
        mask:
        - 720,0,720,27,417,28,409,0
        - 401,310,421,306,431,317,443,336,453,351,459,376,469,398,460,412,442,416,421,411,394,395,364,383,346,372,336,351,345,331,353,314
        - 488,439,518,442,539,461,555,480,568,500,569,522,552,541,533,553,517,559,502,562,484,566,478,532,459,514,437,495,459,479,473,458
        - 720,1006,698,1004,676,989,660,976,620,961,603,938,584,907,567,871,561,850,552,828,570,813,600,809,602,761,619,733,664,733,689,752,720,767,720,769,720,797
```

### Relevant log output

```shell
2023-08-24 06:19:40.918328409  [INFO] Starting Frigate...
2023-08-24 06:19:41.834907889  [2023-08-24 06:19:41] frigate.app                    INFO    : Starting Frigate (0.12.1-367d724)
2023-08-24 06:19:41.865402255  [2023-08-24 06:19:41] peewee_migrate                 INFO    : Starting migrations
2023-08-24 06:19:41.872827003  [2023-08-24 06:19:41] peewee_migrate                 INFO    : There is nothing to migrate
2023-08-24 06:19:41.889449229  [2023-08-24 06:19:41] detector.coral                 INFO    : Starting detection process: 289
2023-08-24 06:19:44.763759606  [2023-08-24 06:19:41] frigate.app                    INFO    : Output process started: 291
2023-08-24 06:19:44.766912976  [2023-08-24 06:19:41] frigate.detectors.plugins.edgetpu_tfl INFO    : Attempting to load TPU as usb
2023-08-24 06:19:44.767051407  [2023-08-24 06:19:41] frigate.app                    INFO    : Camera processor started for holzlager: 297
2023-08-24 06:19:44.767126607  [2023-08-24 06:19:44] frigate.detectors.plugins.edgetpu_tfl INFO    : TPU found
2023-08-24 06:19:44.767186143  [2023-08-24 06:19:41] frigate.app                    INFO    : Camera processor started for werkstatt: 299
2023-08-24 06:19:44.767256939  [2023-08-24 06:19:41] frigate.app                    INFO    : Camera processor started for haustuer_strasse: 300
2023-08-24 06:19:44.767335146  [2023-08-24 06:19:41] frigate.app                    INFO    : Capture process started for holzlager: 302
2023-08-24 06:19:44.767412302  [2023-08-24 06:19:41] frigate.app                    INFO    : Capture process started for werkstatt: 303
2023-08-24 06:19:44.767487132  [2023-08-24 06:19:41] frigate.app                    INFO    : Capture process started for haustuer_strasse: 307
```

### FFprobe output from your camera

```shell
Stream 0:
Return Code: 0

Video:

Codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
Resolution: 3840x2160
FPS: 25/1

Audio:

Codec: AAC (Advanced Audio Coding)

Stream 1:
Return Code: 0


Stream 0:
Return Code: 0

Video:

Codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
Resolution: 3840x2160
FPS: 25/1

Audio:

Codec: AAC (Advanced Audio Coding)

Stream 1:
Return Code: 0


Stream 0:
Return Code: 0

Video:

Codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
Resolution: 2160x3840
FPS: 25/1

Audio:

Codec: AAC (Advanced Audio Coding)

Stream 1:
Return Code: 0
```

### Operating system

UNRAID

### Install method

Docker Compose

### Coral version

USB

### Network connection

Wired

### Camera make and model

3x ANNKE NC800

**Comment by NickM-27** [comment]
This is likely because the database is bloated. The next version of frigate will run vacuum automatically. You can run vacuum manually on the DB file if you'd like.

**Comment by NickM-27** [comment]
I'd suggest making a backup of the DB just in case, but basically:

1. backup db
2. stop frigate
3. change to directory with frigate.db file
4. on the host run `sqlite3 frigate.db`
5. run `VACCUM;`
6. `ctrl + D` to close

**Comment by MasterOfPuppets111** [comment]
Hello,

I have now executed the vacuum command, unfortunately this did not bring any improvement...

I think I listed the command correctly, right? (see image).
![Screenshot_20230824_152727_Samsung Internet](https://github.com/blakeblackshear/frigate/assets/61080924/77983ffe-cdab-4f1d-b650-f2ffb323c8ae)

**Comment by MasterOfPuppets111** [comment]
Hello Nick,
As already said, I'm not a SQL professional, but with ; did it work now the database has shrunk from 40MB to 12MB, but the problem persists....

**Comment by NickM-27** [comment]
Alright, with the problem persisting it just means the DB is very busy and being slow. In that case this has been fixed in 0.13 and you'll need to wait for that 👍

**Comment by MasterOfPuppets111** [comment]
Hello Nick,
Wanted to get back in touch and give feedback that the above issue had absolutely nothing to do with Frigate! The problem was my Unifi Firewall. I had tried everything, even deleted all recordings, completely created a new database, moved the database to my SSD. nothing helped. When I just wanted to delete a recording, a pop-up came to my Unifi Threat Management with the content: "GPL WEB_SERVER DELETE attempt blocked an dropped", starting from the LAN IP of my cell phone from which I wanted to delete the recording. After I classified this IP as trustworthy, deleting it works immediately and Frigate can still be reached. the great side effect: now that my frigate DB is on my cache sdd, the videos load even faster than before!

Thank you again for your help!

Greetings Pierre
